### PR TITLE
Disable Dart's profiler in unoptimized builds

### PR DESCRIPTION
**Pull request by johnmccutchan**

### CODE REVIEW

File: runtime/dart_init.cc [hunk 1]
```diff
@@ -83,6 +83,12 @@ static const char* kDartProfilingArgs[] = {
     // default profile period to 100Hz. This number is suitable for older
     // Raspberry Pi devices but quite low for current smartphones.
     "--profile_period=1000",
+    // Disable Dart's built in profiler when building a debug build. This
+    // works around a race condition that would sometimes stop a crash's
+    // stack trace from being printed on Android.
+#if defined(DEBUG)
```
Comment (jason-simmons) [inline runtime/dart_init.cc]: I think this needs to be #ifndef NDEBUG

**Comment by eseidelGoogle** [comment]
Won't we have this same problem in release?  I guess we'll have to install a fancier crash handler to help our apps and customers apps catch/report crashes as they wish?

File: runtime/dart_init.cc (hunk 1 above)
Comment (johnmccutchan) [inline runtime/dart_init.cc]: Done and thanks -- I was confused about debug vs unoptimized.

**Comment by johnmccutchan** [comment]
@eseidelGoogle No we won't. Release builds ship without a profiler at all.

**Comment by jason-simmons** [comment]
LGTM

**Pull request merged**